packages/xo-web/src/common/xo/index.js

```diff
@@ -1247,10 +1247,18 @@ export const migrateVm = async (vm, host) => {
     return
   }
 
-  if (!params.targetHost) {
+  const { mapVdisSrs, migrationNetwork, sr, targetHost } = params
+
+  if (!targetHost) {
     return error(_('migrateVmNoTargetHost'), _('migrateVmNoTargetHostMessage'))
   }
 
+  // Workaround to prevent VM's VDIs from unexpectedly migrating to the default SR
+  // if migration network is defined, the SR or the map VDI -> SR is required.
+  if (migrationNetwork !== undefined && sr === undefined && mapVdisSrs === undefined) {
```

Comment:
Is `mapVdisSrs === undefined` necessary? If `sr` isn't `undefined`, the VDIs won't be migrated to the default SR anyway, right?